src/object/mapKeys.ts

```diff
@@ -3,28 +3,27 @@
  * by running each own enumerable property of the object through the iteratee function.
  *
  * @template T - The type of the object.
- * @template K1 - The type of the keys in the object.
- * @template K2 - The type of the new keys generated by the iteratee function.
+ * @template K - The type of the new keys generated by the iteratee function.
  *
  * @param {T} object - The object to iterate over.
- * @param {(value: T[K1], key: K1, object: T) => K2} getNewKey - The function invoked per own enumerable property.
- * @returns {Record<K2, T[K1]>} - Returns the new mapped object.
+ * @param {(value: T[keyof T], key: keyof T, object: T) => K} getNewKey - The function invoked per own enumerable property.
+ * @returns {Record<K, T[keyof T]>} - Returns the new mapped object.
  *
  * @example
  * // Example usage:
  * const obj = { a: 1, b: 2 };
  * const result = mapKeys(obj, (value, key) => key + value);
  * console.log(result); // { a1: 1, b2: 2 }
  */
-export function mapKeys<T extends object, K1 extends keyof T, K2 extends PropertyKey>(
+export function mapKeys<T extends Record<PropertyKey, any>, K extends PropertyKey>(
   object: T,
-  getNewKey: (value: T[K1], key: K1, object: T) => K2
-): Record<K2, T[K1]> {
-  const result = {} as Record<K2, T[K1]>;
-  const keys = Object.keys(object);
+  getNewKey: (value: T[keyof T], key: keyof T, object: T) => K
+): Record<K, T[keyof T]> {
+  const result = {} as Record<K, T[keyof T]>;
+  const keys = Object.keys(object) as (keyof T)[];
```

Comment:
I don't know why there is a `K1` type that extends `keyof T`. 

Since there is no reason to extend `keyof T`, we can just use `keyof T`!